Hostid

demmm opened the pull request.

demmm committed:
[zfs] adding hostid
using ZFS in combination with dracut exposed a bug on system updates hostid from the Live session does not match hostid installed, thus zpool id no longer matches id created by dracut in the kernel img
to work around this, the zfs module now uses zgenhostid to create a hostid

demmm committed:
[zfshostid] adding new module
module created to copy zfs generated /etc/hostid, needed to have
hostid in zpool & initramfs in sync

demmm committed:
Changes: add new parts for 3.3.0-aplha3

dalto8 commented on src/modules/zfshostid/zfshostid.schema.yaml:
```diff
@@ -0,0 +1,10 @@
+type: object
+
+required:
+    - zfs
```
I think this means that the item "zfs" is required.  Not the module zfs.  I am not sure that you need a schema for a module which has no config.

I think module requirements are handled by setting `requiredModules` in module descriptor.

dalto8 commented on src/modules/zfshostid/main.py:
```diff
@@ -0,0 +1,45 @@
+        hostid_source = '/etc/hostid'
+        hostid_destination = '{!s}/etc/hostid'.format(root_mount_point)
+
+        if os.path.exists(hostid_source):
```
I am not sure this "if" is needed.  It causes the module to fail without giving an error/warning.

If you remove it, the file being missing should cause an exception which will then write a warning per your exception handler.

dalto8 commented:
I made a couple of small comments on the code itself but I thinks this look good.  I like this approach.

demmm committed:
[zfshostid] simplify run call
required for schema.yaml seems to be for globalstorage entries, not modules
remove section
